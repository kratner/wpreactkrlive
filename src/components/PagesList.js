import React, { Component } from 'react';
import { appGraphQLUrl, gqlPages } from './GetData';

class PagesList extends Component {
  state = {
    pages: []
  };

  componentDidMount() {
    let appUrl = `${appGraphQLUrl}`;
    let queryPages = `${gqlPages}`;
    fetch(appUrl, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: queryPages
    })
      .then(data => data.json())
      .then(data => {
        this.setState({
          pages: data.data.pages.nodes
        });
      });
  }

  render() {
    let listpages = this.state.pages.map((page, index) => {
      return (
        <div key={index}>
          <h4>{page.title}</h4>
          <p
            dangerouslySetInnerHTML={{
              __html: page.content
            }}
          />
          {/* dangerouslySetInnerHTML eliminates the html tags */}
        </div>
      );
    });

    return (
      <div>
        <h1>Pages</h1>
        {listpages}
      </div>
    );
  }
}

export default PagesList;
