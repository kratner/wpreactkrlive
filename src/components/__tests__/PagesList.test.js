import React from "react";
import { shallow } from "enzyme";
import PagesList from "../PagesList";

describe("PagesList", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<PagesList />);
    expect(wrapper).toMatchSnapshot();
  });
});
