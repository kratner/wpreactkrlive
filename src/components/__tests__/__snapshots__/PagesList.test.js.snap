// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagesList renders correctly 1`] = `ShallowWrapper {}`;
